go/chat/server_test.go

```diff
@@ -3631,3 +3632,133 @@ func TestChatSrvTeamChannelNameMentions(t *testing.T) {
+			_verifyHit := func(searchHit chat1.ChatSearchHit) {
+				if prevMessageID == nullMessageID {
+					require.Nil(t, searchHit.PrevMessage)
+				} else {
+					require.True(t, searchHit.PrevMessage.IsValid())
+					require.EqualValues(t, searchHit.PrevMessage.Valid().MessageID, prevMessageID)
+				}
+				require.EqualValues(t, searchHit.HitMessage.Valid().MessageID, hitMessageID)
+				require.Equal(t, searchHit.Matches, matches)
+
+				if nextMessageID == nullMessageID {
+					require.Nil(t, searchHit.NextMessage)
+				} else {
+					require.True(t, searchHit.NextMessage.IsValid())
+					require.EqualValues(t, searchHit.NextMessage.Valid().MessageID, nextMessageID)
+				}
+
+			}
+			_verifyHit(searchHit)
+			select {
+			case searchHitArg := <-searchHitCb:
+				_verifyHit(searchHitArg.SearchHit)
+			case <-time.After(20 * time.Second):
+				require.Fail(t, "no search result received")
+			}
+		}
+		verifySearchDone := func(numHits int) {
+			select {
+			case searchDone := <-searchDoneCb:
+				require.Equal(t, searchDone.NumHits, numHits)
+			case <-time.After(20 * time.Second):
+				require.Fail(t, "no search result received")
+			}
+		}
+
+		search := func(query string, maxHits int, maxMessages int) chat1.GetSearchRegexpRes {
+			res, err := tc.chatLocalHandler().GetSearchRegexp(tc.startCtx, chat1.GetSearchRegexpArg{
+				ConversationID: conv.Id,
+				Query:          query,
+				MaxHits:        maxHits,
+				MaxMessages:    maxMessages,
+			})
+			require.NoError(t, err)
+			return res
+		}
+
+		maxHits := 1
+		maxMessages := 1000
+		// Test basic equality match
+		query := "hi"
+		msgBody := "hi"
+		messageID := sendMessage(msgBody)
+		res := search(query, maxHits, maxMessages)
+		require.Equal(t, len(res.Hits), 1)
+		verifyHit(nullMessageID, messageID, nullMessageID, []string{query}, res.Hits[0])
+		verifySearchDone(1)
+
+		// Test basic no results
+		query = "hey"
+		res = search(query, maxHits, maxMessages)
+		require.Equal(t, len(res.Hits), 0)
+		verifySearchDone(0)
+
+		// Test maxHits
+		maxHits = 1
+		query = "hi"
+		messageID1 := sendMessage(msgBody)
+		res = search(query, maxHits, maxMessages)
+		require.Equal(t, len(res.Hits), 1)
+		verifyHit(messageID, messageID1, nullMessageID, []string{query}, res.Hits[0])
+		verifySearchDone(1)
+
+		maxHits = 5
+		res = search(query, maxHits, maxMessages)
+		require.Equal(t, len(res.Hits), 2)
+		verifyHit(messageID, messageID1, nullMessageID, []string{query}, res.Hits[0])
+		verifyHit(nullMessageID, messageID, messageID1, []string{query}, res.Hits[1])
+		verifySearchDone(2)
+
+		messageID2 := sendMessage(msgBody)
+		res = search(query, maxHits, maxMessages)
+		require.Equal(t, len(res.Hits), 3)
+		verifyHit(messageID1, messageID2, nullMessageID, []string{query}, res.Hits[0])
+		verifyHit(messageID, messageID1, messageID2, []string{query}, res.Hits[1])
+		verifyHit(nullMessageID, messageID, messageID1, []string{query}, res.Hits[2])
+		verifySearchDone(3)
+
+		// Test regex functionality
```

Comment:
Might want to also test escaped regexes (`"a\.b"`).

Reply:
added!